packages/core-transaction-pool-mem/lib/mem.js

```diff
@@ -154,6 +171,12 @@ class Mem {
       delete this.bySender[senderPublicKey]
     }
 
+    const recipientId = memPoolTransaction.transaction.recipientId
+    this.byRecipient[recipientId].delete(memPoolTransaction)
+    if (this.byRecipient[recipientId].size === 0) {
+      delete this.byRecipient[recipientId]
+    }
+
```

Comment:
Removal happens in reverse order to addition. E.g. add to member1, add to member2, add to member3. Remove from member3, remove from member2, remove from member1. That is, move this new code to line 168.

Reply:
ok please check